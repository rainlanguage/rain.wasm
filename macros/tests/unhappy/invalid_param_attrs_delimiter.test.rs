#[macro_use]
extern crate wasm_bindgen_utils_macros;

struct TestStruct;

#[wasm_export]
impl TestStruct {
    #[wasm_export]
    pub fn test_method1(
        #[wasm_export(param_description = "first"; param_description = "second")]
        arg: String,
    ) -> Result<String, Error> {
        Ok(arg)
    }
}

#[wasm_export] 
impl TestStruct {
    #[wasm_export]
    pub fn test_method2(
        #[wasm_export(param_description = "test" - something_else)]
        arg: String,
    ) -> Result<String, Error> {
        Ok(arg)
    }
}

fn main() {}