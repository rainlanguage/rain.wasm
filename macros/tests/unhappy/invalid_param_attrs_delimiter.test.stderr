error: expected `,`
  --> tests/unhappy/invalid_param_attrs_delimiter.test.rs:10:50
   |
10 |         #[wasm_export(param_description = "first"; param_description = "second")]
   |                                                  ^

error: expected string literal
  --> tests/unhappy/invalid_param_attrs_delimiter.test.rs:21:23
   |
21 |         #[wasm_export(param_description = "test" - something_else)]
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
